jazz_environments/components/validation/validate_create_payload.js

```diff
@@ -23,168 +23,234 @@
-        },
-        // Check if a service with same domain and service_name combination exists
-        validateServiceExists: function(onComplete) {
-            logger.info("validateServiceExists: ");
-            var service_domain = environment_data.domain;
-            var service_name = environment_data.service;
-            var svcGetPayload;
-            //call services get
-            svcGetPayload = {
-                uri: global.config.SERVICE_API_URL + global.config.SERVICE_API_RESOURCE + "?domain=" + service_domain + "&service=" + service_name,
-                method: "GET",
-                headers: {'Authorization':global.authorization},
-                rejectUnauthorized: false
-            };
-            request(svcGetPayload, function(error, response, body) {
-                if (response.statusCode === 200) {
-                    var output = JSON.parse(body);
-                    if (!output.data || !output.data || output.data.available == null) {
-                        onComplete({
-                            result: "inputError",
-                            message: "Error finding service: " + service_domain + "." + service_name + " in service catalog"
-                        }, null);
-                    } else if (output.data.available === false) {
-                        onComplete(null, {
-                            result: "success",
-                            message: "Service is available!"
-                        });
-                    } else if (output.data.available === true) {
-                        onComplete({
-                            result: "inputError",
-                            message: "Service with domain: " + service_domain + " and service name:" + service_name + ", does not exist."
-                        },null);
-                    }
-                } else {
-                    onComplete({
+function validateAllRequiredFieldsValue(environment_data, required_fields_create) {
+    //check for required fields
+    logger.info("Inside validateAllRequiredFieldsValue: ");
+    return new Promise((resolve, reject) => {
+        validateUtils.validateAllRequiredFieldsValue(environment_data, required_fields_create, function onValidate(error, data){
+            if(error){
+                reject(error);
+            } else{
+                resolve(data);
+            }
+        });
+    });
+};
+
+function validateRemoveEmptyValues(environment_data) {
+    // check for empty values before updating environments table
+    logger.info("Inside validateRemoveEmptyValues: ");
+    return new Promise((resolve, reject) =>{
+        validateUtils.validateRemoveEmptyValues(environment_data, function onValidate(error, data){
+            if(error){
+                reject(error);
+            } else{
+                resolve(data);
+            }
+        });
+    });
+};
+
+function validateServiceExists(environment_data) {
+    logger.info("validateServiceExists: ");
```

Comment:
This is at best debug. I would remove it.

Reply:
using debug